File: Python/Level 2/위장 (해시).py
def solution(clothes):
    cloth = {}
    cnt = 1

    for i in clothes:
        if i[1] in cloth:
            cloth[i[1]] += 1
        else:
            cloth[i[1]] = 1
    
    for i in cloth:
        cnt *= (cloth[i] + 1)

    # 옷을 입는 모든 가짓수는 (종류 개수+1)의 곱에서 모두 안 입는 경우인 1을 제거
    return cnt - 1

File: Javascript/Level 2/위장 (해시).js
function solution(clothes) {
	let answer = 1;
	let dic = {};

	clothes.forEach((v) => {
		let sort = v[1];

		if (v[1] in dic) {
			dic[v[1]] += 1;
		} else {
			dic[v[1]] = 1;
		}
	});

	// 모든 조합의 개수는 (조합+1)의 곱 - 1
	for (let v of Object.keys(dic)) {
		answer *= dic[v] + 1;
	}

	return answer - 1;
}
